### ISSUE TITLE
Mutable and other serializers cannot deserialize different content_types

### ISSUE THREAD

**Issue by shumphrey** [PerlDancer/Dancer2#973]
Since 598ffa7922014b47158954a87e1b90f3516e26bb Dancer2::Serializer::Mutable (or other serializers) is no longer able to deserialize multiple content_types.

Previously, when Dancer2::Core::Request was about to deserialize, it first made a call to Serializer->support_content_type($ct)
Mutable took this opportunity to set its own content type to that so that when it came to deserialize, it knew the content type.
This is obviously quite hacky but I assume it worked (I've not tested that version).

https://github.com/PerlDancer/Dancer2/blob/8d544721bf9e706cdfb9c160ab75a63a9c579a99/lib/Dancer2/Core/Request.pm#L288

Currently, Mutable cannot deserialize anything but the default content type application/json

I would suggest the the serializer be passed the request's content_type as a parameter in the deserialize method.

**Comment by veryrusty** [comment]
Not that it helps, but my memory is that the Mutable serializer has "issues" before that too. There are some notes in #795 which also have a linked PR that was incomplete.

**Comment by shumphrey** [comment]
True, but in my case I don't use Mutable, I use a home grown thing that does a similar job, and it cannot do the right thing because it isn't given enough information.
As far as I can see, the only thing deserialize needs is the content_type. I don't think it will ever need anything else? I'll knock up a PR that passes request->content_type as an option when it deserializes.

**Comment by shumphrey** [comment]
So yes, PR #805 and issue #795 are all about this.
